src/hugr.rs

```diff
@@ -66,10 +66,10 @@ pub struct NodeType {
 
 impl NodeType {
     /// Create a new optype with some ExtensionSet
-    pub fn new(op: impl Into<OpType>, input_extensions: ExtensionSet) -> Self {
+    pub fn new(op: impl Into<OpType>, input_extensions: impl Into<Option<ExtensionSet>>) -> Self {
```

Comment:
I don't think the `Into` is necessary - I'm not sure what we would want to put here that implements the interface

Reply:
This is because the library provides a blanket implementation of `Into<Option<T>>` for all T (using `Some`). So using that with `T`==`ExtensionSet`, this generalizes the old `new`.